### PR TITLE
Project team organization user model relationships

### PR DESCRIPTION
**Pull request by begedin**
# Description

Add the entities "Team", "Organization" and "Project" and interlinks them, as well as linking them with users.
- A `user` has multiple `projects`, can belong to multiple `teams` through the `team_members` table, and can belong to multiple `organizations` through the `organization_members` table. 
- The `organization_members` table additionally has a `role` column, which can be either `"admin"` or `"regular"`. I'm guessing the `team_members` table will end up having it also.
- An `organization` can own multiple `teams` and `projects` and has multiple `members` through the `team_members` table
- A `team` belongs to an `organization` and has multiple `members` through the `team_members` table.
- A `project` belongs to an `owner`, which can be a `user` or an `organization` (polymorphic)

### CODE REVIEW

**Comment by begedin** [comment]
Not sure why build is failing. Works on my machine

@JoshSmith I'm gonna need codeship access so I can check.

**Comment by joshsmith** [comment]
@begedin the issue was with the `config/database.yml`. Your local settings were overriding Codeship's. You need to add to `.gitignore` typically.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
```
Comment (joshsmith) [inline .editorconfig:9]: @begedin you know my friend made `.editorconfig`, right?

**Comment by joshsmith** [comment]
Really good work. :+1:

File: .editorconfig (hunk 1 above)
Comment (begedin) [inline .editorconfig:9]: Seriously? Did not know that. Cool!

Comment (begedin) [inline .editorconfig:9]: Also, thank him for me, please :)

**Pull request merged**